### PR TITLE
feat: add scale transform

### PR DESCRIPTION
**Pull request by jscanvic**
In this PR, we suggest adding the scale transform [1] to the existing shift and rotate transforms. 

[1] https://arxiv.org/abs/2312.11232

### Checks to be done before submitting your PR
- [ ] `python3 -m pytest tests/` runs successfully.
- [x] `black .` runs successfully.
- [ ] `make html` runs successfully (in the `docs/` directory).
- [ ] Updated docstrings related to the changes (as applicable).
- [x] Added an entry to the [CHANGELOG.rst](../CHANGELOG.rst).

### CODE REVIEW

File: deepinv/transform/scale.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    N = torch.tensor(len(values), device=device, dtype=dtype)
+    indices = torch.floor(N * torch.rand(shape, device=device, dtype=dtype)).to(
+        torch.int
+    )
+    return values[indices]
+
+
+class Scale(Module):
+    """
+    2D Scaling
```
Comment (matthieutrs) [inline deepinv/transform/scale.py:18]: Could you make a short description of what "2D scaling" is, before adding ref to paper? Then you can also add pointers to this function in [https://deepinv.github.io/deepinv/deepinv.loss.html#transforms](https://deepinv.github.io/deepinv/deepinv.loss.html#transforms)? (i.e. add an entry in [here](https://github.com/deepinv/deepinv/blob/main/docs/source/deepinv.loss.rst))

**Review by matthieutrs — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! Could you please add the appropriate tests, probably similar to [that](https://github.com/deepinv/deepinv/blob/main/deepinv/tests/test_loss.py#L41C62-L41C62)?

**Comment by jscanvic** [comment]
Thanks for reviewing the PR! It should be good now.

**Review by matthieutrs — APPROVED** [review]
